### DISCUSSION TITLE
No CSS when creating component from Java class

### DISCUSSION THREAD

**Discussion by EvaldasMin** [primefaces/community#658] [PrimeFaces] [answered]
I'm getting no style class when creating component in Java class. For example when creating Command Button and placing it inside panel its created without any CSS class. When same action is made inside XHTML newly created component has these CSS classes: ui-button ui-widget ui-state-default ui-corner-all ui-button-text-only

```java
CommandButton cmdButton = new CommandButton();
cmdButton.setTitle("Just Test");
uiPanelGrid.getChildren().add(cmdButton);
```

**Comment by tandraschko** [comment]
You need to use the JSF Application#createComponent API, Not just the constructor of the component.

**Comment by EvaldasMin** [comment]
When I'm using JSF Application#createComponent API, I still end up with component without any style.

```java
FacesContext context = FacesContext.getCurrentInstance();
UIComponent composite = context.getApplication().getViewHandler()
	  .getViewDeclarationLanguage(context, context.getViewRoot().getViewId())
	  .createComponent(context, "http://primefaces.org/ui", "commandButton", null);
uiPanelGrid.getChildren().add(composite);
```
Is there a way to programmatically create Primefaces component with its default / theme style?

**Reply by tandraschko** [reply]
You dont use it correctly, pf components are no composites

**Reply by EvaldasMin** [reply]
When I'm using it likes this component is still rendered with empty style / CSS class property

```java
Application application = FacesContext.getCurrentInstance().getApplication();
CommandButton cmdButton = (CommandButton) application.createComponent(CommandButton.COMPONENT_TYPE);
cmdButton.setTitle("Just test");
uiPanelGrid.getChildren().add(cmdButton);
```

**Reply by tandraschko** [reply]
Then we need a primefaces-test reproducer

**Reply by EvaldasMin** [reply]
I'm attaching it.
[primefaces-test-reproducer.zip](https://github.com/primefaces/community/files/13644156/primefaces-test-reproducer.zip)
**Re-uploaded correct version**

**Reply by melloware** [reply]
Your test reproducer has nothing in it?  It looks like the default reproducer did you attach the correct zip?

**Reply by EvaldasMin** [reply]
> Your test reproducer has nothing in it?  It looks like the default reproducer did you attach the correct zip?
> 

I re-uploaded zip.

**Reply by tandraschko** [reply, marked as answer]
why one button uses title, the other uses value?
commandButton is built to use value || icon
